pkg/util/modules/modules_test.go

```diff
@@ -47,3 +47,109 @@ func TestDependencies(t *testing.T) {
 	assert.Nil(t, svcs)
 	assert.Error(t, err, fmt.Errorf("unrecognised module name: service_unknown"))
 }
+
+func TestRegisterModuleWithOptions(t *testing.T) {
+	sut := NewManager()
+	sut.RegisterModule("module1", mockInitFunc)
+
+	m := sut.modules["module1"]
+
+	assert.NotNil(t, mockInitFunc, m.initFn, "initFn not assigned")
+	assert.True(t, m.public, "module should be public")
+}
+
+func TestRegisterModuleSetsDefaultOption(t *testing.T) {
+	sut := NewManager()
+	sut.RegisterModule("module1", mockInitFunc)
+
+	m := sut.modules["module1"]
+
+	assert.True(t, m.public, "mould should be public")
+}
+
+func TestFunctionalOptAtTheEndWins(t *testing.T) {
+	publicMod := func(option *module) {
+		option.public = true
+	}
+	sut := NewManager()
+	sut.RegisterModule("mod1", mockInitFunc, PrivateModule, publicMod, PrivateModule)
+
+	m := sut.modules["mod1"]
+
+	assert.NotNil(t, mockInitFunc, m.initFn, "initFn not assigned")
+	assert.False(t, m.public, "module should be public")
+}
+
+func TestGetAllPublicModulesNames(t *testing.T) {
+	sut := NewManager()
+	sut.RegisterModule("public1", mockInitFunc)
+	sut.RegisterModule("public2", mockInitFunc)
+	sut.RegisterModule("public3", mockInitFunc)
+	sut.RegisterModule("private1", mockInitFunc, PrivateModule)
+	sut.RegisterModule("private2", mockInitFunc, PrivateModule)
+
+	pm := sut.PublicModuleNames()
+
+	assert.Len(t, pm, 3, "wrong result slice size")
+	assert.Contains(t, pm, "public1", "missing public module")
+	assert.Contains(t, pm, "public2", "missing public module")
+	assert.Contains(t, pm, "public3", "missing public module")
+}
+
+func TestGetAllPublicModulesNamesHasNoDupWithDependency(t *testing.T) {
+	sut := NewManager()
+	sut.RegisterModule("public1", mockInitFunc)
+	sut.RegisterModule("public2", mockInitFunc)
+	sut.RegisterModule("public3", mockInitFunc)
+
+	assert.NoError(t, sut.AddDependency("public1", "public2", "public3"))
+
+	pm := sut.PublicModuleNames()
+
+	// make sure we don't include any module twice because there is a dependency
+	assert.Len(t, pm, 3, "wrong result slice size")
+	assert.Contains(t, pm, "public1", "missing public module")
+	assert.Contains(t, pm, "public2", "missing public module")
+	assert.Contains(t, pm, "public3", "missing public module")
+}
+
+func TestGetEmptyListWhenThereIsNoPublicModule(t *testing.T) {
+	sut := NewManager()
+	sut.RegisterModule("private1", mockInitFunc, PrivateModule)
+	sut.RegisterModule("private2", mockInitFunc, PrivateModule)
+	sut.RegisterModule("private3", mockInitFunc, PrivateModule)
+	sut.RegisterModule("private4", mockInitFunc, PrivateModule)
+
+	pm := sut.PublicModuleNames()
+
+	assert.Len(t, pm, 0, "wrong result slice size")
+}
+
+func TestPublicModuleNamesReturnsSortedList(t *testing.T) {
```

Comment:
If suggestion in `TestGetAllPublicModulesNames` is accepted, this test can be dropped.